exercicios/para-casa/entregas/valenthina-temudo/bankAccount.js

```diff
@@ -0,0 +1,127 @@
+const Client = require('./client');
+const Bank = require('./bank');
+
+class BankAccount {
+  client;
+  bank;
+  accountNumber;
+  agencyNumber;
+  #amount = 0;
+  #qtyCashWithdrawals24h = 0;
+  #withdrawalRate24h = 1.5;
+
+  constructor(client, bank, accountNumber, agencyNumber) {
+    if (client instanceof Client) {
+      if (bank instanceof Bank) {
+        const isClientBank = client.associatedBanks.includes(bank);
+        if (isClientBank) {
+          this.client = client;
+          this.bank = bank;
+          this.accountNumber = accountNumber;
+          this.agencyNumber = agencyNumber;
+        }
+      } else {
+        console.log('Bank must be a instanceof Bank');
+      }
+    } else {
+      console.log('Client must be a instanceof Client');
+    }
+  }
+
+  get amount() {
+    return this.#amount;
+  }
+
+  set amount(newAmount) {
+    this.#amount = newAmount;
+  }
+
+  get qtyCashWithdrawals24h() {
+    return this.#qtyCashWithdrawals24h;
+  }
+
+  set qtyCashWithdrawals24h(newQtyCashWithdrawals24h) {
+    this.#qtyCashWithdrawals24h = newQtyCashWithdrawals24h;
+  }
+
+  get withdrawalRate24h() {
+    return this.#withdrawalRate24h;
+  }
+
+  set withdrawalRate24h(newWithdrawalRate24h) {
+    this.#withdrawalRate24h = newWithdrawalRate24h;
+  }
+
+  credit(amount) {
+    this.#amount += amount;
+    console.log(`Deposit successfully of ${amount}. Balance: ${this.#amount}`);
+  }
+
+  debit(amount) {
+    this.#amount -= amount;
+    console.log(
+      `Withdrawal successfully of ${amount}. Balance: ${this.#amount}`
+    );
+  }
+
+  transferTo(anotherAccount, amount) {
+    if (anotherAccount instanceof BankAccount) {
+      if (this.#amount < amount) {
```

Comment:
Se o valor que estou tentando transferir for maior do que o saldo da conta, ele retorna um console.log avisando "Insufficient balance for transfer", mas ainda permite a transação porque não encerra a função. Uma solução poderia ser colocar return ao invés do console.log, para fazer essa finalização.